glean-core/src/storage/mod.rs

```diff
@@ -143,4 +143,92 @@ impl StorageManager {
 
         snapshot
     }
+
+    ///  Snapshot the experiments.
+    ///
+    /// ## Arguments:
+    ///
+    /// * `storage` - The database to get data from.
+    /// * `store_name` - The store name to look into.
+    ///
+    /// ## Return value
+    ///
+    /// Returns a JSON representation of the experiment data, in the following format:
+    ///
+    /// {
+    ///  'experiment-id': {
+    ///    'branch': 'branch-id',
+    ///    'extra': {
+    ///      'additional': 'property',
+    ///      // ...
+    ///    }
+    ///  }
+    /// }
+    ///
+    /// Returns `None` if no data for experiments exists.
+    pub fn snapshot_experiments_as_json(
+        &self,
+        storage: &Database,
+        store_name: &str,
+    ) -> Option<JsonValue> {
+        let mut snapshot: HashMap<String, HashMap<String, JsonValue>> = HashMap::new();
+
+        let mut snapshotter = |metric_name: &[u8], metric: &Metric| {
+            let metric_name = String::from_utf8_lossy(metric_name).into_owned();
+            if metric_name.ends_with("#experiment") {
+                let name = metric_name.splitn(2, '#').next().unwrap(); // safe unwrap, first field of a split always valid
+                let map = snapshot
+                    .entry(metric.category().into())
+                    .or_insert_with(HashMap::new);
+                map.insert(name.to_string(), metric.as_json());
+            }
+        };
+
+        // FIXME: glean_internal_info should probably not be known here or, at least, shared.
+        storage.iter_store_from(Lifetime::Application, store_name, &mut snapshotter);
+
+        if snapshot.is_empty() {
+            None
+        } else {
+            Some(json!(snapshot))
```

Comment:
```suggestion
            Some(json!({"experiments": snapshot}))
```

Reply:
Wait, do we need the outer object? The new doc comment says no, but we previously had it, so it depends on how it is handled when inserted into the ping payload.

Reply:
Let's remove the outer `"experiments"`, I like your suggestion more :)